src/overlays/actors/ovl_En_Kujiya/z_en_kujiya.c

```diff
@@ -35,48 +44,331 @@ const ActorInit En_Kujiya_InitVars = {
     (ActorFunc)EnKujiya_Draw,
 };
 
-#endif
+#define CHECK_LOTTERY_NUMBERS                                                \
+    (((u32)((void)0, gSaveContext.lotteryCodes[CURRENT_DAY - 1][0]) ==       \
+      ((((void)0, gSaveContext.lotteryCodeGuess & 0xFFFF) & 0xF00) >> 8)) && \
+     ((u32)((void)0, gSaveContext.lotteryCodes[CURRENT_DAY - 1][1]) ==       \
+      ((((void)0, gSaveContext.lotteryCodeGuess & 0xFFFF) & 0xF0) >> 4)) &&  \
+     ((u32)((void)0, gSaveContext.lotteryCodes[CURRENT_DAY - 1][2]) ==       \
+      (((void)0, gSaveContext.lotteryCodeGuess & 0xFFFF) & 0xF)))
 
-extern UNK_TYPE D_06006198;
+void EnKujiya_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnKujiya* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kujiya/EnKujiya_Init.s")
+    Actor_SetScale(&this->actor, 0.1f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kujiya/EnKujiya_Destroy.s")
+    this->actor.flags &= ~ACTOR_FLAG_1;
+    this->actor.targetMode = 6;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kujiya/func_80BB09A8.s")
+    this->actor.focus.pos = this->actor.world.pos;
+    this->actor.focus.pos.y += 30.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Kujiya/func_80BB09BC.s")
+    if (func_80BB0FF8() && !(gSaveContext.time < CLOCK_TIME(6, 0)) && (gSaveContext.time < CLOCK_TIME(18, 0))) {
```

Comment:
```suggestion
    if (func_80BB0FF8() && (gSaveContext.time >= CLOCK_TIME(6, 0)) && (gSaveContext.time < CLOCK_TIME(18, 0))) {
```
This should also work and is clearer imo.